rfd/0221-mcp-gateway.md

```diff
@@ -0,0 +1,129 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD 0221 - MCP Gateway
+
+## Required Approvers
+
+* Engineering: @r0mant && @greedy52
+* Product: @klizhentas
+
+## What
+
+A single local MCP server that can serve all Teleport MCPs to clients,
+providing a single experience for managing configurations and troubleshooting
+issues.
+
+## Why
+
+The MCP local gateway reduces the operational overhead of setting up each
+resource by allowing clients to integrate once instead of editing configurations
+for each new server. It complements the current approach while improving
+visibility of runtime failures, providing actionable errors, and centralizing
+remediation in Teleport Connect/`tsh`.
+
+## UX
+
+The MCP gateway is available in both tsh and Teleport Connect. This section
+mainly focuses on integration within Teleport Connect, with a brief subsection
+on MCP gateway usage in `tsh` at the end.
+
+### Guiding users into knowing Teleport's MCP capabilities and features
+
+Previously, Teleport Connect did not support launching or connecting to MCP
+applications. Now, users will see a "Connect" button on MCP applications and an
+additional connect option for supported databases. By clicking it, they will
+enable the server on the gateway.
+
+<img width="871" alt="Unified Resources view in Connect with MCP support" src="assets/0221-resources-view.png">
```

Comment:
Nitpicking here, but instead of using the App icons as the main icon of the MCP and the MCP icon as a subheader, can we just use the MCP as the new icon type for them? Using the sub icon, for example, to indicate that a DB is MCP compatible, makes sense. From a UX perspective, I would want to be able to quickly visually parse MCPs in a list of available MCPs.